mathesar/api/db/viewsets/records.py

```diff
@@ -60,43 +57,42 @@ def list(self, request, table_pk=None):
                 field='grouping',
                 status_code=status.HTTP_400_BAD_REQUEST
             )
-
-        serializer = RecordSerializer(records, many=True)
+        serializer = RecordSerializer(
+            records,
+            many=True,
+            context=self.get_serializer_context(table)
+        )
         return paginator.get_paginated_response(serializer.data)
 
     def retrieve(self, request, pk=None, table_pk=None):
         table = get_table_or_404(table_pk)
         record = table.get_record(pk)
         if not record:
             raise NotFound
-        serializer = RecordSerializer(record)
+        serializer = RecordSerializer(record, context=self.get_serializer_context(table))
         return Response(serializer.data)
 
     def create(self, request, table_pk=None):
         table = get_table_or_404(table_pk)
         # We only support adding a single record through the API.
-        assert isinstance((request.data), dict)
-        try:
-            record = table.create_record_or_records(request.data)
-        except IntegrityError as e:
-            if type(e.orig) == NotNullViolation:
-                raise database_api_exceptions.NotNullViolationAPIException(
-                    e,
-                    status_code=status.HTTP_400_BAD_REQUEST,
-                    table=table
-                )
-            else:
-                raise database_api_exceptions.MathesarAPIException(e, status_code=status.HTTP_400_BAD_REQUEST)
-        serializer = RecordSerializer(record)
+        serializer = RecordSerializer(data=request.data, context=self.get_serializer_context(table))
+        serializer.is_valid(raise_exception=True)
```

Comment:
I think this would be clearer as an `if` statement.

Reply:
Also applies below.

Reply:
Most of the codebase seems to not use an `if` statement when calling `is_valid`, so I followed a similar approach but using an `if` statement is my preferred way of writing, would love to make that change in other places too along with this change to maintain consistency.